src/internal/Subscription.ts

```diff
@@ -166,6 +166,8 @@ export class Subscription implements SubscriptionLike {
           const tmp = subscription;
           subscription = new Subscription();
           subscription._subscriptions = [tmp];
+        } else if (this._subscriptions && this._subscriptions.some(s => s === subscription)) {
+          return subscription;
         }
```

Comment:
I've reviewed the failing tests and you are correct: your changes to `Subscription` are needed. However, I'd like them to be done a little differently.

I'd like the change you've made to the `add` method in `Subscription.ts` to be reverted and, instead, I'd like the check to be moved to `_addParent` - so that all of the parent checks are co-located. Like this:

```ts
  /** @internal */
  private _addParent(parent: Subscription): boolean {
    let { _parent, _parents } = this;
    if (_parent === parent) {
      // If the new parent is the same as the current parent, then do nothing.
      return false;
    } else if (!_parent) {
      // If we don't have a parent, then set this._parent to the new parent.
      this._parent = parent;
      return true;
    } else if (!_parents) {
      // If there's already one parent, but not multiple, allocate an Array to
      // store the rest of the parent Subscriptions.
      this._parents = [parent];
      return true;
    } else if (_parents.indexOf(parent) === -1) {
      // Only add the new parent to the _parents list if it's not already there.
      _parents.push(parent);
      return true;
    }
    return false;
  }
```

Reply:
I have implemented your requested changes and also added you as collaborator to the repository fork.